41.md

```diff
@@ -0,0 +1,46 @@
+NIP-41
+======
+
+### Poll & Vote Event
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+An event with kind `1` that wants to contain a poll can do so by adding poll tags and its content as follows:
+
+```json
+{
+  ...
+  "kind": 1,
+  "tags": [
+    ...
+    ["poll", "Choice 1"],
+    ["poll", "Choice 2"],
+    ["poll", "Choice 3"]
+    ...
+  ],
+}
+```
+
+Then the client should display voting options if more than **1 choice** is present in the poll.
+
+The client can then submit the vote by sending an event with kind `8` and its content which will be the index (start at `0`) of the chosen choice.
```

Comment:
Why not re-use/extend/abuse the NIP-25 reactions (type `7`).

Then the poll options instead would be:
```json
{
    "tags": [
        ["poll", "Choice 1"],
        ["poll", "Choice 2"],
        ["poll", "Choice 3"],
    ]
}
```

and reaction/response event type `7` would be like this:

```json
{
    "id": "<ID>",
    "sig": "<signature>",
    "kind": 7,
    "tags": [
        ["e", "<event ID of the note with the poll>"],
        ["p", "<pubkey of the note with the poll>"]
    ],
    "pubkey": "<pubkey of the responder>",
    "content": "Choice 1",
    "created_at": 1234567891
}
```

Where the `"content"` has to exactly match the choices given to count, otherwise it's treated like a normal reaction.

For better interoperability( /not breaking the current type `7` use), using 1️⃣, 2️⃣, 3️⃣, 4️⃣ etc. as first char of the `"poll"`tag and then allowing the reaction to only need to include the minimum unique prefix to 'count' so a simple 1️⃣ emoji reaction would work as a vote:

**poll**

```json
{
    "tags": [
        ["poll", "1️⃣ Choice 1 text"],
        ["poll", "2️⃣ Choice 2 text"],
        ["poll", "3️⃣ Choice 3 text"],
    ]
}
```

**reaction**

```json
{
    ...,
    "content": "1️⃣",
}
```

Reply:
I did see on your Damus PR though that @jb55 wasn't sure about re-using/overloading the reaction type

Reply:
The problem with using reactions is that the reaction is not necessarily used to respond to a poll but is intended for "reaction" use only.
So with @jb55, we thought it would be better to explain this "feature" via a dedicated NIP :)

Reply:
Polls are an event, reactions to an event that is a poll is indeed an answer, seems perfect simple to me. YAGNI!